components/idp-mgt/org.wso2.carbon.idp.mgt/src/main/java/org/wso2/carbon/idp/mgt/dao/IdPManagementFacade.java

```diff
@@ -183,12 +186,12 @@ public void deleteIdP(String idPName, int tenantId, String tenantDomain)
             throws IdentityProviderManagementException {
 
         IdentityProvider identityProvider = getIdPByName(null, idPName, tenantId, tenantDomain);
-        deleteEndpointConfig(identityProvider, tenantDomain);
+        dao.deleteIdP(idPName, tenantId, tenantDomain);
         try {
-            dao.deleteIdP(idPName, tenantId, tenantDomain);
+            deleteEndpointConfig(identityProvider, tenantDomain);
         } catch (IdentityProviderManagementException e) {
-            addEndpointConfig(identityProvider, tenantDomain);
-            throw e;
+            // Error will not be thrown, since the IDP is already deleted. But there will be stale action.
+            LOG.warn(String.format(IdPManagementConstants.WarningMessage.WARN_STALE_IDP_ACTION, idPName));
```

Comment:
Since IdP is deleted just printing the IdP name will not help on finding stale actions right ? IMO, we need to log the action id referenced with the deleted IdP as well